Mathlib/Data/BitVec/Defs.lean

```diff
@@ -140,4 +140,12 @@ def toLEList (x : BitVec w) : List Bool :=
 def toBEList (x : BitVec w) : List Bool :=
   List.ofFn x.getMsb'
 
+instance : SMul ℕ (BitVec w) := ⟨fun x y => ofFin <| x • y.toFin⟩
+instance : SMul ℤ (BitVec w) := ⟨fun x y => ofFin <| x • y.toFin⟩
+instance : Pow (BitVec w) ℕ  := ⟨fun x n => ofFin <| x.toFin ^ n⟩
+instance : NatCast (BitVec w) := ⟨BitVec.ofNat w⟩
+
+/-- The obvious instance would be to define `IntCast` in terms of `Std.BitVec.ofInt`.  -/
+instance : IntCast (BitVec w) := ⟨fun x => ofFin x⟩
```

Comment:
Using `.ofInt` probably is the right thing to do, since ultimately `ofInt` is likely to be more efficient.

Reply:
Fair point, I've redefined `IntCast` and proven the necessary equivalence with `Fin`s `IntCast` instance